packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3512,7 +3518,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       if (!widget.readOnly) {
         _scheduleShowCaretOnScreen(withAnimation: true);
       }
-      if (!_value.selection.isValid) {
+      if (kIsWeb && !_isMultiline && !_nextFocusChangeIsInternal) {
```

Comment:
Should we add `&& selectionEnabled` to this condition? If we want to disable selection for an input, we probably don't want users to be able to circumvent that via tab